Source/WebCore/platform/SharedBuffer.h

```diff
@@ -367,6 +353,7 @@ class SharedBufferBuilder {
     {
         ensureBuffer();
         Ref { *m_buffer }->append(std::forward<Args>(args)...);
+        m_buffer->m_contiguous = m_buffer->segmentsCount() <= 1;
```

Comment:
Ditto here, it seems like m_buffer would set m_contiguous itself and not the builder.